Sources/TuistLoader/Models+ManifestMappers/TestAction+ManifestMapper.swift

```diff
@@ -9,22 +9,51 @@ extension TuistCore.TestAction {
     ///   - manifest: Manifest representation of test action model.
     ///   - generatorPaths: Generator paths.
     static func from(manifest: ProjectDescription.TestAction, generatorPaths: GeneratorPaths) throws -> TuistCore.TestAction {
-        let targets = try manifest.targets.map { try TuistCore.TestableTarget.from(manifest: $0, generatorPaths: generatorPaths) }
-        let arguments = manifest.arguments.map { TuistCore.Arguments.from(manifest: $0) }
-        let configurationName = manifest.configurationName
-        let coverage = manifest.coverage
-        let codeCoverageTargets = try manifest.codeCoverageTargets.map {
-            TuistCore.TargetReference(projectPath: try generatorPaths.resolveSchemeActionProjectPath($0.projectPath),
-                                      name: $0.targetName)
+        let testPlans: [TuistCore.TestPlan]?
+        let targets: [TuistCore.TestableTarget]
+        let arguments: TuistCore.Arguments?
+        let coverage: Bool
+        let codeCoverageTargets: [TuistCore.TargetReference]
+        let diagnosticsOptions: Set<TuistCore.SchemeDiagnosticsOption>
+        let language: String?
+        let region: String?
+
+        if let plans = manifest.testPlans {
```

Comment:
It's a good time to create a `TestPlan+ManifestMapper.swift` and move all this logic there

Reply:
please check again, I think that doesn't make sense anymore.